src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -112,6 +118,19 @@ public final void prepareRun(BatchSinkContext context) throws Exception {
 
   @Override
   public final void onRunFinish(boolean succeeded, BatchSinkContext context) {
+    if (succeeded) {
+      Job queryJob = bigQuery.getJob(jobId);
+      if (queryJob == null) {
+        LOG.warn("Unable to find job, metrics cannot be collected");
+      } else {
```

Comment:
nit: in general, it's more readable to have less nesting and do these types of checks like:

```
if (!succeeded) {
  return;
}

Job queryJob = bigQuery.getJob(jobId);
if (queryJob == null) {
  LOG.warn(...);
  return;
}

// other logic here.
```

though you'll have to move this block to its own method or move it below the existing file delete logic.

Reply:
Done.